scalableminds/webknossos
# Mesh follow up

### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- open a dataset with a mesh (let me know if you need that)
- move around, request a precomputed mesh, delete the mesh from the list (ideally, all of this should happen while data is loading; I did this by throttling the Internet speed)

### Issues:
- fixes #5506 with the following points:
  - [x] when reading segment id from centered voxel, also take into account fallback buckets if necessary. this also makes sure to correctly deal with fallback loading while "render missing data black" is enabled (previously, the code always backed out if the setting was enabled even though fallback rendering is at work when the bucket is still being fetched)
  - [x] load chunks in parallel to improve loading speed. now, a sliding window is applied to process the jobs with 6 "workers". this can clog the request queue of the entire page, but this should be alright, since the user requested the meshes explicitly. the reason I chose (and built) a pool for this is that the request can be easily aborted then. in hindsight (I'm just noticing that), aborting the requests via an `AbortController` would have been alright, too :thinking:  However, now, we can tweak the parallelization degree better.
  - [x] don't fetch mesh for segment 0
    - [x] more generic: don't crash if the list-request fails / returns empty
  - [x] deleting a segment while it's still loading produces an invalid state
- addtionally, the coloring of the meshes was tweaked a bit

------
